### PR TITLE
[F001]-create_classes_and_prepare_data

### PR DESCRIPTION
**Pull request by mdunggggg**
# Summary:
Introduced new employee-related classes for NhanVien, NhanVienVanPhong, NhanVienLapTrinh, and QuanLy.
# Changes:
#### Created the following classes:
- [x] NhanVien: Represents a generic employee with common attributes.
- [x] NhanVienVanPhong: Represents an office employee, inherits from NhanVien.
- [x] NhanVienLapTrinh: Represents a programming employee, inherits from NhanVien.
- [x] QuanLy: Represents a manager
# Checklist I have done
- [x] Prepared data
- [x] Added classes.
- [x] Conducted basic testing for each class.
- [x] Verified there are no conflicts with existing components.
# Result
Result

### CODE REVIEW

**Review by Tris1702 — CHANGES_REQUESTED** [review]
Please push your code in a specific folder (such as: ManhDung)

**Review by Tris1702 — CHANGES_REQUESTED** [review]
Please provide a description for this merge request, some information you should provide such as:
- What is this MR about?
- Checklist of what you have done in this MR
- How can reviewers test your code? (Manual for running code only. If there are some specific steps, please describe more details)
- Image of results

Thanks, happy coding

File: .idea/.gitignore
Comment (Tris1702) [inline .idea/.gitignore:1]: don't push .idea folder! (this folder is generated when you using IntelliJ, without them, I still can run code => so .idea folder is not important)

File: BTVN-Buoi-1.iml
Comment (Tris1702) [inline BTVN-Buoi-1.iml:1]: don't push this file (same reason)

**Review by Tris1702 — CHANGES_REQUESTED** [review]
Please remove not important files

File: NhanVien.dart [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class NhanVien{
+  static int maNVCount = 1;
+  String maNV = '${maNVCount++}';
+  String hoTen;
+  String ngaySinh;
+  String soDT;
+  String diaChi;
+  double luong;
+  NhanVien(this.hoTen, this.ngaySinh, this.soDT, this.diaChi, this.luong);
```
Comment (VuNA2k2) [inline NhanVien.dart]: I recommend using named parameters in this case. It makes your code cleaner.

File: NhanVienKinhDoanh.dart [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import 'NhanVien.dart';
+
+class NhanVienKinhDoanh extends NhanVien {
+    int doanhSo;
+    NhanVienKinhDoanh(String hoTen, String ngaySinh, String soDT, String diaChi, double luong, this.doanhSo) : super(hoTen, ngaySinh, soDT, diaChi, luong){
```
Comment (VuNA2k2) [inline NhanVienKinhDoanh.dart]: Same issue.

File: NhanVienLapTrinh.dart [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import 'NhanVien.dart';
+
+class NhanVienLapTrinh extends NhanVien{
+  int level;
+  NhanVienLapTrinh(String hoTen, String ngaySinh, String soDT, String diaChi, double luong, this.level) : super(hoTen, ngaySinh, soDT, diaChi, luong){
```
Comment (VuNA2k2) [inline NhanVienLapTrinh.dart]: Same issue

**Review by VuNA2k2 — APPROVED** [review]

**Review by Tris1702 — APPROVED** [review]

File: Data.dart [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+class Data{
+  List<NhanVien> getData(){
+    List<NhanVien> dsNV = [];
+    dsNV.add(NhanVienLapTrinh(hoTen: 'Nguyen Van A', ngaySinh: '01/01/1990', soDT: '0123456789', diaChi: 'Ha Noi', luong: 1000, level: 1));
+    dsNV.add(NhanVienLapTrinh(hoTen: 'Nguyen Van B', ngaySinh: '01/01/1990', soDT: '0123456789', diaChi: 'Ha Noi', luong: 1000, level: 2));
+    dsNV.add(NhanVienLapTrinh(hoTen: 'Nguyen Van C', ngaySinh: '01/01/1990', soDT: '0123456789', diaChi: 'Ha Noi', luong: 1000, level: 3));
+    dsNV.add(NhanVienLapTrinh(hoTen: 'Nguyen Van E', ngaySinh: '01/01/1990', soDT: '0123456789', diaChi: 'Ha Noi', luong: 1000, level: 4));
+
+
+    dsNV.add(NhanVienKinhDoanh(hoTen: 'Nguyen Van D', ngaySinh: '01/01/1990', soDT: '0123456789', diaChi: 'Ha Noi', luong: 1000, doanhSo: 100));
```
Comment (ProJulius) [inline Data.dart:14]: test review

**Review by ghost — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Review by Wos2610 — APPROVED** [review]

**Pull request merged**